spec/hashie/mash_spec.rb

```diff
@@ -824,4 +824,21 @@ class SubMash < Hashie::Mash
       end
     end
   end
+
+  with_minimum_ruby('2.5.0') do
+    describe '#slice' do
+      subject { described_class.new(a: 'apple', b: 'banana') }
+      it 'accepts both string and symbol as key' do
+        expect(subject.slice(:a)).to eq('a' => 'apple')
+        expect(subject.slice('a')).to eq('a' => 'apple')
+      end
+
+      it 'returns an instance of the class it was called on' do
+        class KlassMash < Hashie::Mash; end
+        klass_mash = KlassMash.new('a' => 'apple', 'b' => 'banana')
+
+        expect(klass_mash.slice('a')).to be_a(KlassMash)
```

Comment:
For completeness:

```suggestion
        expect(klass_mash.slice('a')).to eq KlassMash.new(a: 'apple')
        expect(klass_mash.slice(:a)).to eq KlassMash.new(a: 'apple')
```

Reply:
I check the hash equality in the test on line 831, this test I want to make sure it's an instance of whatever class called it, since `KlashMash.new(a: 'apple') == { 'a' => 'apple' }` would return `true`

Reply:
Oh, good point. I agree that checking the class is important, I just forgot how equality worked. With subclassing, we sometimes have bugs where the subclass doesn't stick (😞) and we sometimes have bugs where the behavior changes (😓). I was trying to capture both the class check and behavior in one go.